docs/rfd.md

```diff
@@ -0,0 +1,478 @@
+  // system is configured.
+  repeated string subjects = 3;
+}
+
+```
+
+The RBAC configuration will be simply loaded from a config file on server startup. An example config file might look like:
+
+```yaml
+roles:
+  - id: adminRole
+    service: job.v1.Job
+    allowedMethods:
+      - Start
+      - Stop
+      - Status
+      - List
+      - Output
+roleBindings:
+  - id: adminRoleBinding
+    roleId: adminRole
+    subjects:
+      - admin
+```
+
+APIs and message types are versioned, initially starting at v1. This will allow future major versions to be introduced without breaking existing clients. Adding new fields to existing messages can be done without breaking compatibility, as long as existing field tags are not changed.
+
+### UX
+
+#### API
+
+The methods of the Job service API are designed such that each method accomplishes a single task and requires the least amount of stateful knowledge as possible.
+
+Other than `Start()` and `List()`, which are fully stateless, the only stateful parameter required for interacting with the various methods is a Job ID. Job IDs are completely opaque to the user, which reduces client complexity. Furthermore, the IDs are automatically given to the user upon starting a job, so the client has all the information it needs in order to perform any other operation for that job.
+
+Example usage of the GRPC API would look like:
+
+1. Starting a job:
+
+```go
+var client jobv1.JobClient
+// ...
+mcores := int64(1000) // 1 core
+jobId, err := client.Start(cmd.Context(), &jobv1.JobSpec{
+  Command: &jobv1.CommandSpec{
+    Command: "kubectl",
+    Args:    []string{"logs", "--follow", "pod/example"},
+    Env:     []string{"KUBECONFIG=/path/to/kubeconfig"},
+  },
+  Limits: &jobv1.ResourceLimits{
+    Cpu: &mcores,
+  },
+})
+```
+
+2. Streaming output:
+
+```go
+stream, err := client.Output(cmd.Context(), &jobv1.OutputRequest{
+  Id:     jobId,
+  Follow: true,
+})
+if err != nil {
+  return err
+}
+for {
+  resp, err := stream.Recv()
+  if err != nil {
+    return err
+  }
+  fmt.Fprint(os.Stdout, resp.GetOutput())
+}
+```
+
+3. Stopping the job:
+
+```go
+_, err := client.Stop(ctx, jobId)
+```
+
+#### CLI
+
+The CLI is the primary means of interacting with the Job Server. It will have a familiar interface to other CLI tools like `docker` or `kubectl`.
+The following examples demonstrate how the CLI will be used:
+
+```bash
+# starting the job server
+$ sudo jobserver serve --listen-address=127.0.0.1:9090 --rbac=rbac-config.yaml \
+  --cacert=ca.crt --cert=tls.crt --key=tls.key
+
+# to use the client, flags to configure cert and key paths, and the server address, are required.
+# these are omitted in the remaining examples for brevity.
+$ jobserver --address=127.0.0.1:9090 --cacert=ca.crt --cert=tls.crt --key=tls.key ...
+
+# starting a job:
+# for commands that don't require flag args, they can be passed as-is:
+$ jobserver run kubectl logs pod/example
```

Comment:
Nit: It's a little strange that both server and client are the same binary `jobserver`. Think it would make sense to have a separate tool like `jobctl` for the client?

Reply:
Sure, I can do that. :+1:  Building everything into one binary is usually my personal preference (although it is a bit unorthodox), but it probably doesn't make much of a difference in this case.